### PR TITLE
test(store): ensure store for plugins

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: packages/store/tests/ensure-store.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  class CountState {
+    @Action({ type: 'increment' })
+    public addOne(): void {}
+
+    @Action({ type: 'increment' })
+    public addTwo(): void {}
+  }
+
+  beforeAll(async () => {
+    await TestBed.configureTestingModule({
```
Comment (arturovt) [inline packages/store/tests/ensure-store.spec.ts]: This is not needed. `compileComponents` is only needed if you pass `templateUrl` to the `@Component`.

File: packages/store/tests/ensure-store.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      imports: [NgxsModule.forRoot([CountState])]
+    }).compileComponents();
+  });
+
+  it('should be get meta data from state', () => {
```
Comment (markwhitfeld) [inline packages/store/tests/ensure-store.spec.ts]:
```suggestion
  it('should get the meta data from the state', () => {
```

File: packages/store/tests/ensure-store.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { TestBed } from '@angular/core/testing';
+import { State, NgxsModule, ensureStoreMetadata, Action } from '@ngxs/store';
+
+describe('Ensure store for plugins', () => {
```
Comment (arturovt) [inline packages/store/tests/ensure-store.spec.ts]: Ensure store metadata?

Comment (arturovt) [inline packages/store/tests/ensure-store.spec.ts]:
```suggestion
describe('Ensure store metadata', () => {
```

Comment (arturovt) [reply packages/store/tests/ensure-store.spec.ts]: Better name?

Comment (markwhitfeld) [reply packages/store/tests/ensure-store.spec.ts]: Yes, I think so

**Comment by splincode** [comment]
@markwhitfeld @arturovt please see it

File: packages/store/tests/ensure-store.spec.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        }
+
+        const metadata: SelectorMetaDataModel = getSelectorMetadata(
+          SuperCountState.canInheritSelectFn
+        );
+
+        expect(metadata.containerClass).toEqual(SuperCountState);
+      } catch (e) {
+        // TODO(splincode): is normal for SuperCountState?
+        expect(extractFunctionBody(e.message)).toEqual(
```
Comment (arturovt) [inline packages/store/tests/ensure-store.spec.ts]: Could you create a local variable and assign it in the `catch` block:
```ts
let error: Error | null = null;

try {
  ...
} catch (e) {
  error = e;
}

expect(error).toBeDefined();
expect(extractFunctionBody....)...
```

File: packages/store/tests/ensure-store.spec.ts [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+      expect(metadata.originalFn).not.toEqual(CountState.selectFn);
+      expect(extractFunctionBody(metadata.originalFn)).toEqual(
+        'function (state) { return state; }'
+      );
+
+      expect(metadata.originalFn!(1)).toEqual(1); // state => state
+
+      expect(extractFunctionBody(metadata.getSelectorOptions)).toEqual(
+        'function () { return getCustomSelectorOptions(selectorMetaDataClone, getExplicitSelectorOptions()); }'
```
Comment (arturovt) [inline packages/store/tests/ensure-store.spec.ts]: Isn't it dangerous to compare with the string here? I mean if we change any functionality of the `getSelectorOptions` in the future we will need to always fix this test.

File: packages/store/tests/ensure-store.spec.ts (hunk 4 above)
Comment (splincode) [reply packages/store/tests/ensure-store.spec.ts]: done

File: packages/store/tests/ensure-store.spec.ts (hunk 5 above)
Comment (splincode) [reply packages/store/tests/ensure-store.spec.ts]: done

**Pull request merged**